rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/activityList.spec.ts

```diff
@@ -373,22 +402,62 @@ describe("ActivityList", () => {
   });
 
   it("fetches data automatically when auto-refresh is true", async () => {
-    // to do: add all 3 mocks for axiosMock (since, reports, execution), to ensure that loadSince will pass;
-    axiosMock.get.mockResolvedValue({
-      data: {
-        reports: reports,
-      },
+    // Use fake timers
+    jest.useFakeTimers();
+    // Mock axios.get for different API calls
+    axiosMock.get.mockImplementation((url) => {
+      if (url.includes("eventsAjax")) {
+        return Promise.resolve({
+          data: {
+            total: reports.length,
+            reports: reports,
+          },
+        });
+      }
+      if (url.includes("running")) {
+        return Promise.resolve({
+          data: {
+            executions: [],
+          },
+        });
+      }
+      if (url.includes("since.json")) {
+        return Promise.resolve({
+          data: {
+            since: {
+              count: 0,
+            },
+          },
+        });
+      }
+      return Promise.resolve({ data: {} });
```

Comment:
@smartinellibenedetti I have use all three API's. Can you please review if I did it correctly